softlayer/cloud-config.yml

```diff
@@ -40,19 +39,20 @@ networks:
     dns: [((internal_ip)), 8.8.8.8, 10.0.80.11, 10.0.80.12]
     static: ((internal_static_ips))
     reserved: ((internal_reserved_ips))
+    cloud_properties:
+      vlan_ids:
+        ((sl_vlan_public))
+        ((sl_vlan_private))
 
 - name: dynamic
   type: dynamic
   subnets:
   - dns: [((internal_ip)), 8.8.8.8, 10.0.80.11, 10.0.80.12]
     azs: [z1, z2, z3]
     cloud_properties:
-      PrimaryNetworkComponent:
-        NetworkVlan:
-          Id: ((sl_vlan_public))
-      PrimaryBackendNetworkComponent:
-        NetworkVlan:
-          Id: ((sl_vlan_private))
+      vlan_ids:
+        ((sl_vlan_public))
+        ((sl_vlan_private))
```

Comment:
Just curious... does the CPI internally figure out which of this is the public vs private VLAN? Or is that something the SoftLayer API figures out itself?

Reply:
We added a logic in CPI NG code to distinguish which one is pri or pub. For user experience, we simplified the definition in yml.